core/execution/future/market.go

```diff
@@ -997,6 +997,31 @@ func (m *Market) parkAllPeggedOrders(ctx context.Context) []*types.Order {
 	return parked
 }
 
+func (m *Market) UpdateMarketState(ctx context.Context, changes *types.MarketStateUpdateConfiguration) error {
+	if changes.UpdateType == types.MarketStateUpdateTypeTerminate {
+		// terminate and settle
+		m.tradingTerminated(ctx, true)
+		n := num.Numeric{}
+		n.SetUint(changes.SettlementPrice)
+		m.settlementData(ctx, &n)
```

Comment:
Do we not want to set the market state to `CLOSED`? In the market lifecycle spec its distinct from `SETTLE`:
https://github.com/vegaprotocol/specs/blob/cosmicelevator/protocol/0043-MKTL-market_lifecycle.md#closed

And also a protos value that already exists for it ready an waiting:
```
// Governance vote to close (Not currently implemented)
    STATE_CLOSED = 7;
```

Theres also mention of uncrossing auctions before closing?

Reply:
idk, because we're settling the market at least for future so not sure this is still as was originally designed. It's a question for @witgaw, same about the uncrossing auctions.

Reply:
`STATE_CLOSED` would be ideal and yes, we'd want to uncross any auctions there may be first - will clarify both these points in the spec.

Reply:
done